arch/riscv/core/smp.c

```diff
@@ -74,12 +76,17 @@ void arch_secondary_cpu_init(int hartid)
 #endif
 #ifdef CONFIG_SMP
 	irq_enable(RISCV_IRQ_MSOFT);
+#endif
+#ifdef CONFIG_SOC_SMP_PER_CORE_INIT_HOOK
+	soc_smp_per_core_init_hook();
```

Comment:
We should run this for the main core, probably by adding an invocation of this in [`arch_kernel_init()` similar to the `z_riscv_pmp_init()`](https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/riscv/include/kernel_arch_func.h#L54), so that the function is ran by the main core as well, otherwise this would be a `per_secondary_core_init_hook`

I also think that it would be nice to generalize it into `soc_per_core_init_hook()` so that it is not SMP-specific. So that any soc that would like some additional core initialization code can use this hook